src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -59,97 +61,97 @@ public override void VisitMarkupTextLiteral(MarkupTextLiteralSyntax node)
 
     public override void VisitMarkupLiteralAttributeValue(MarkupLiteralAttributeValueSyntax node)
     {
-        AddSemanticRange(node, RazorSemanticTokensLegend.MarkupAttributeQuote);
+        AddSemanticRange(node, _razorSemanticTokensLegend.MarkupAttributeQuote);
     }
 
     public override void VisitMarkupAttributeBlock(MarkupAttributeBlockSyntax node)
     {
         Visit(node.NamePrefix);
-        AddSemanticRange(node.Name, RazorSemanticTokensLegend.MarkupAttribute);
```

Comment:
For methods in this class that access `_razorSemanticTokensLegend` multiple times, consider pulling it into a local variable first to avoid multiple field loads.

Reply:
This is a level of micro-optimization I have not previously considered, or achieved 😀